# Construction of pYPK0_TEF1_PsXYL1_TDH3 (pMEC1126)


[pYPKa_Z_TEF1](pYPKa_Z_TEF1.ipynb)

[pYPKa_E_TDH3](pYPKa_E_TDH3.ipynb) 

In [25]:
import pydna
from pydna import ipynb_importer

In [26]:
from ypk_std_primers import p166, p167, p567, p568

[Yeast Pathway Kit Standard Primers](ypk_std_primers.ipynb)

In [27]:
from Bio.Restriction import ZraI, AjiI, EcoRV

In [28]:
pYPK0 = pydna.read("pYPK0.gb")

In [29]:
promoter_clone = pYPKa_Z_TEF1 = pydna.read("pYPKa_Z_TEF1.gb")

[Genbank](http://www.ncbi.nlm.nih.gov/nuccore/X59465.1) record

In [30]:
from IPython.display import HTML
HTML('<iframe src=http://www.ncbi.nlm.nih.gov/pubmed/1756986 width=100% height=250></iframe>')

In [31]:
gb = pydna.Genbank("bjornjobbb@gmail.com")
gene_template = gb.nucleotide("X59465.1")

In [32]:
gene_template.list_features()

+----------+-----------+-------+------+--------+--------------+--------------+------+
| Feature# | Direction | Start | End  | Length | id           | type         | orf? |
+----------+-----------+-------+------+--------+--------------+--------------+------+
| 0        |    -->    |   0   | 2052 |   2052 | <unknown id> | source       |  no  |
| 1        |    -->    |   0   | 355  |    355 | <unknown id> | regulatory   |  no  |
| 2        |    -->    |  192  | 199  |      7 | <unknown id> | regulatory   |  no  |
| 3        |    -->    |  272  | 281  |      9 | <unknown id> | regulatory   |  no  |
| 4        |    -->    |  287  | 301  |     14 | <unknown id> | stem_loop    |  no  |
| 5        |    -->    |  342  | 1366 |   1024 | <unknown id> | misc_feature |  no  |
| 6        |    -->    |  355  | 1312 |    957 | <unknown id> | gene         | yes  |
| 7        |    -->    |  355  | 1312 |    957 | <unknown id> | CDS          | yes  |
| 8        |    -->    |  1334 | 1339 |      5 | <unkn

In [33]:
orf = gene_template.extract_feature(7)

In [34]:
orf.isorf()

True

In [35]:
print len(orf)
orf.seguid()

957


lGWKlZ3EDAyY2V-7tTQoJRqgW6w

In [36]:
print str(orf.seq.translate())

MPSIKLNSGYDMPAVGFGCWKVDVDTCSEQIYRAIKTGYRLFDGAEDYANEKLVGAGVKKAIDEGIVKREDLFLTSKLWNNYHHPDNVEKALNRTLSDLQVDYVDLFLIHFPVTFKFVPLEEKYPPGFYCGKGDNFDYEDVPILETWKALEKLVKAGKIRSIGVSNFPGALLLDLLRGATIKPSVLQVEHHPYLQQPRLIEFAQSRGIAVTAYSSFGPQSFVELNQGRALNTSPLFENETIKAIAAKHGKSPAQVLLRWSSQRGIAIIPKSNTVPRLLENKDVNSFDLDEQDFADIAKLDINLRFNDPWDWDKIPIFV*


In [37]:
pf, pr = pydna.parse('''>XR_YPK_rec_fwd 
                        gtcgaggaacgccaggttgcccactttctcactagtgaAAATGCCTTCTATTAAGTTGAACTCTGG
                        >XR_YPK_rec_rev 
                        ATTTAAatcctgatgcgtttgtctgcacagatggcgcgTTAGACGAAGATAGGAATCTTGTCCC''')

In [38]:
g = pydna.pcr( pf, pr, orf)

In [39]:
g.figure()

                                        5ATGCCTTCTATTAAGTTGAACTCTGG...GGGACAAGATTCCTATCTTCGTCTAA3
                                                                      |||||||||||||||||||||||||| tm 54.0 (dbd) 66.4
                                                                     3CCCTGTTCTAAGGATAGAAGCAGATTgcgcggtagacacgtctgtttgcgtagtcctaAATTTA5
5gtcgaggaacgccaggttgcccactttctcactagtgaAAATGCCTTCTATTAAGTTGAACTCTGG3
                                         |||||||||||||||||||||||||| tm 53.2 (dbd) 64.7
                                        3TACGGAAGATAATTCAACTTGAGACC...CCCTGTTCTAAGGATAGAAGCAGATT5

In [40]:
terminator_clone = pYPKa_E_TDH3 = pydna.read("pYPKa_E_TDH3.gb")

In [41]:
p  = pydna.pcr( p167, p567, promoter_clone)
t  = pydna.pcr( p568, p166, terminator_clone)

In [42]:
pYPK0_E_Z, stuffer = pYPK0.cut((EcoRV, ZraI))

In [43]:
(pYPK0_E_Z, p, g, t)

(Dseqrecord(-5681), Dseqrecord(-741), Dseqrecord(-1035), Dseqrecord(-804))

In [44]:
asm = pydna.Assembly((pYPK0_E_Z, p, g, t), limit=31)

In [45]:
asm

Assembly:
Sequences........................: [5681] [741] [1035] [804]
Sequences with shared homologies.: [5681] [741] [804] [1035]
Homology limit (bp)..............: 31
Number of overlaps...............: 4
Nodes in graph(incl. 5' & 3')....: 6
Only terminal overlaps...........: No
Circular products................: [8015]
Linear products..................: [8113] [8076] [8065] [8052] [7422] [7309] [7067] [6424] [6324] [2493] [1802] [1726] [98] [61] [50] [37]

In [46]:
candidate = asm.circular_products[0]
candidate.figure()

 -|pYKP0|98
|        \/
|        /\
|        98|741bp_PCR_prod|38
|                          \/
|                          /\
|                          38|1035bp_PCR_prod|35
|                                             \/
|                                             /\
|                                             35|804bp_PCR_prod|61
|                                                               \/
|                                                               /\
|                                                               61-
|                                                                  |
 ------------------------------------------------------------------

In [47]:
pYPK0_TEF1_PsXYL1_TDH3 = candidate.synced(pYPK0)

The pYPK0_TEF1_PsXYL1_TDH3 has cseguid ```ak608NPPLpMmn_YIHGwQPcQzee0``` and 8015 bp

In [48]:
print len(pYPK0_TEF1_PsXYL1_TDH3)
pYPK0_TEF1_PsXYL1_TDH3.cseguid()

8015


ak608NPPLpMmn_YIHGwQPcQzee0

In [49]:
pYPK0_TEF1_PsXYL1_TDH3.name="pMEC1126"
#pYPK0_TEF1_PsXYL1_TDH3.description = "pYPK0_TEF1_PsXYL1_TDH3tp"
pYPK0_TEF1_PsXYL1_TDH3.stamp()

cSEGUID_ak608NPPLpMmn_YIHGwQPcQzee0_2015-07-15T12:33:22.383260

In [50]:
pYPK0_TEF1_PsXYL1_TDH3.write("pYPK0_TEF1_PsXYL1_TDH3.gb")

###[pYPK0_TEF1_PsXYL1_TDH3](pYPK0_TEF1_PsXYL1_TDH3.gb)

##Download sequence

[pYPK0_TEF1_PsXYL1_TDH3](pYPK0_TEF1_PsXYL1_TDH3.gb)

In [51]:
reloaded = pydna.read("pYPK0_TEF1_PsXYL1_TDH3.gb")

In [52]:
reloaded.cseguid()

ak608NPPLpMmn_YIHGwQPcQzee0